packages/edit-site/src/components/global-styles/ui.js

```diff
@@ -326,6 +344,7 @@ function GlobalStylesUI() {
 
 			<GlobalStylesActionMenu />
 			<GlobalStylesBlockLink />
+			<GlobalStylesRevisionLink />
```

Comment:
I think we might need to feed some more context to this hook. It seems to prevent switching between revisions and the stylebook.


https://github.com/WordPress/gutenberg/assets/6458278/e3cbb268-8ab0-49e3-8b89-e227885113a6

My assumption is that `editorCanvasContainerView === 'global-styles-revisions'` is already `true` so the panel doesn't switch.

Maybe we need something like `canvasMode === 'edit'` or something?

Reply:
Oh, good catch! I'll look into it, but yes, hopefully something like `canvasMode === 'edit'` might work.

Reply:
I think I've figured it out @ramonjd — it seems that the `useEffect` was firing correctly (only when the container is updated to `global-styles-revisions`), but that the Style Book button didn't appear to play nicely when the path was explicitly set to `/revisions`. As a fix, I've updated this `useEffect` so that it also handles redirects for when the canvas container is updated to `style-book`, by redirecting to the root.

I believe this is needed because the style book button lives outside of the navigation provider for global styles, so it's probably good for consistency for us to handle the redirects in the one place.

I've updated in https://github.com/WordPress/gutenberg/pull/51149/commits/51e26bc9e1ad17963f71ceea40b4afae29602013 and https://github.com/WordPress/gutenberg/pull/51149/commits/8ae9c7ab372ed677d6370bc077e37828c87e2c36 — the latter ensures that we only redirect to the root path when we're on the revisions screen. I _think_ the revisions screen is the only one where the style book won't work nicely, and this ensures that if someone is deeply nested within the global styles screens, opening the style book doesn't unexpectedly redirect back to root.

Let me know if you think this is an okay direction or if we should go in a different route... so to speak! 😄

Reply:
> Let me know if you think this is an okay direction or if we should go in a different route... so to speak!

I'm sure we can navigate towards a good solution :trollface:

Reply:
Zing! 🤣